editor/plugins/node_3d_editor_plugin.cpp

```diff
@@ -2252,6 +2252,11 @@ void Node3DEditorViewport::_sinput(const Ref<InputEvent> &p_event) {
 		}
 
 		if (_edit.mode == TRANSFORM_NONE) {
+			if (!_edit.gizmo.is_valid() && is_freelook_active()) {
```

Comment:
```suggestion
			if (_edit.gizmo.is_null() && is_freelook_active()) {
```

Reply:
This is missing a check for `k->get_keycode() == Key::ESCAPE` too. Right now, pressing *any* key will exit freelook.

Reply:
Thanks!